src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -90,38 +98,103 @@ def for_cli(cls, ws: WorkspaceClient, product='ucx'):
     def index(self):
         return self._migration_status_refresher.index()
 
-    def migrate_tables(self, *, what: What | None = None, acl_strategy: list[AclMigrationWhat] | None = None):
+    def migrate_tables(self, *, what: What, acl_strategy: list[AclMigrationWhat] | None = None):
+        if what in [What.DB_DATASET, What.UNKNOWN]:
+            logger.error(f"Can't migrate tables with type { what.name }")
+            return None
+        all_grants_to_migrate = None if acl_strategy is None else self._gc.snapshot()
+        all_migrated_groups = None if acl_strategy is None else self._group.snapshot()
+        all_principal_grants = None if acl_strategy is None else self._principal_grants.get_interactive_cluster_grants()
         self._init_seen_tables()
         tables_to_migrate = self._tm.get_tables_to_migrate(self._tc)
-        tasks = []
-        if acl_strategy is not None:
-            grants_to_migrate = self._gc.snapshot()
-            migrated_groups = self._group.snapshot()
-            principal_grants = self._principal_grants.get_interactive_cluster_grants()
+        all_tasks = []
+        if what == What.VIEW:
```

Comment:
```suggestion
        if what == What.VIEW:
            return self._migrate_views(...)
```
let's keep the diff to this method to minimum